bitcoin/src/bip32.rs

```diff
@@ -112,40 +112,76 @@ pub struct Xpub {
 #[cfg(feature = "serde")]
 crate::serde_utils::serde_string_impl!(Xpub, "a BIP-32 extended public key");
 
+/// A non-hardened child number.
+///
+/// # Note
+///
+/// Key index, within [0, 2^31 - 1].
+#[derive(Copy, Clone, PartialEq, Eq, Debug, PartialOrd, Ord, Hash)]
+pub struct NormalChildNumber(pub u32);
```

Comment:
Since this type is supposed to maintain the invariant that the top bit is not set, IMO the inner field should be private. Same for the hardened struct.